modules/runtime-fields-common/src/main/java/org/elasticsearch/runtimefields/RuntimeFieldsPainlessExtension.java

```diff
@@ -28,6 +28,11 @@
 public class RuntimeFieldsPainlessExtension implements PainlessExtension {
     private final List<Whitelist> whitelists;
 
+    // we don't use ServiceProvider directly, but module-info wants this
+    public RuntimeFieldsPainlessExtension() {
```

Comment:
I wonder if we should just remove the META-INF/services from this component, since it’s not necessary (and causes the module build sanity check to fail)

Reply:
Then we don’t need this no args constructor.

Reply:
I think that adding the no-args ctr here is ok for now. It unblocks the modularization effort. We can revisit the extension points as a future follow up.